cmd/finch/nerdctl.go

```diff
@@ -474,3 +477,43 @@ var nerdctlCmds = map[string]string{
 	"volume":    "Manage volumes",
 	"wait":      "Block until one or more containers stop, then print their exit codes",
 }
+
+var managementCmds = []string{
```

Comment:
Are management commands the only type of commands that can have sub-arguments?

Reply:
just to make sure i understood correctly
nerdctl compose up
in this case this is a management command as up is not an option, is this what the question about?
Note: compose is not part of this list.

Reply:
The fix in this PR is to change the way Finch handles parsing local env and then passing it into the VM context via the limactl/nerdctl command. Currently, when Finch does this, it takes a command like `finch run -it --env="E=e" alpine:latest` and transforms it into `finch run --env="E=e" -it alpine:latest`, because `run` is identified as the end of the "command".

This PR fixes the parsing logic by adding a new list of commands that have subcommands (the `managementCmds`), so that if you run a command like `finch container run --debug -it --rm --env="E=v" busybox env`, it recognizes `container run` as a single command, and adds the `--env` after `container run`, instead of after `container`.

A better approach may be to entirely rework how environment variables are handled, by not re-arranging where `--env` is in the command, and instead replacing `--env` in place.

So in the example in the PR description,

```
finch container run --debug -it --rm --env="E=v" busybox env
```

becomes

```
shell finch sudo -E nerdctl container run  -it --rm --env="E=v" busybox env
```

instead of

```
shell finch sudo -E nerdctl container run -e E=v -it --rm busybox env
```

Reply:
sure in that case let me raise a fresh cr and once that checks out we can cancel this one